google-cloud-clients/google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/Publisher.java

```diff
@@ -430,7 +431,21 @@ public void shutdown() throws Exception {
    * <p>Call this method to make sure all resources are freed properly.
    */
   public boolean awaitTermination(long duration, TimeUnit unit) throws InterruptedException {
-    return publisherStub.awaitTermination(duration, unit);
+    messagesWaiter.waitNoMessages();
+    boolean isAwaited = publisherStub.awaitTermination(duration, unit);
+    if (isAwaited) {
+      for (AutoCloseable closeable : closeables) {
+        ExecutorAsBackgroundResource executorAsBackgroundResource =
+            (ExecutorAsBackgroundResource) closeable;
+        isAwaited = executorAsBackgroundResource.awaitTermination(duration, unit);
```

Comment:
The user is asking for something like: wait for 8 seconds total for termination.

`publisherStub.awaitTermination(duration, unit);` could take 2.1 seconds.  The next await termination should wait 8-2.1 seconds.

Please keep track of the start time, and the number of millis that have gone by since the method was invoked, and then wait the remainder of the users requested timeout.

Reply:
@sduskis changes done